lib/rows.js

```diff
@@ -99,6 +99,13 @@ function queryRowsImpl(self, builtPlan, streamType, options, bindingArg, graphql
 }
 
 function queryRowsOperationImpl(self, builtPlan, streamType, options, bindingArg, graphqlQuery) {
+  console.log('line 102')
+ // console.log(builtPlan['_fn'])
+  console.log(options)
+  let updateFunctions = new Set().add('write').add('lock-for-update').add('remove');
```

Comment:
This won't work for Optic DSL statements, as those are strings submitted to the server and we don't want to try parsing those out to look for specific functions.

Per the design in what is now ticket MLE-4118, we need the user to explicitly tell us that they are performing an update. I think asking the user to include "update=true" in `options` is an easy approach.